From dc0b680699ec67ace0825724087487c3f5f82886 Mon Sep 17 00:00:00 2001
From: Yu Watanabe <watanabe.yu+github@gmail.com>
Date: Sat, 18 Mar 2023 12:17:54 +0900
Subject: [PATCH] core/transaction: make merge_unit_ids() return non-NULL on
 success

(cherry picked from commit 999f16514367224cbc50cb3ccc1e4392e43f6811)

Related: RHEL-132317
---
 src/core/transaction.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/core/transaction.c b/src/core/transaction.c
index 89cc1d0a4c..cd80e7e111 100644
--- a/src/core/transaction.c
+++ b/src/core/transaction.c
@@ -341,6 +341,9 @@ static char* merge_unit_ids(const char* unit_log_field, char **pairs) {
                 size += next + 1;
         }
 
+        if (!ans)
+                return strdup("");
+
         return TAKE_PTR(ans);
 }
 
